File: C++/Courses/Course-4/Melih/Teacher.cpp
#include "Teacher.h"

Teacher::Teacher(string name, int age, string department):Person(name, age){
    this->department = department;
}

void Teacher::printInfo(){
    cout<<"Name: "<< name<< endl;
    cout<<"Age: "<< age<< endl;
    cout<<"Department: "<< department<< endl; 
}

File: Java/Courses/Course-3/src/OOPII/Inheritance/Teacher.java
package OOPII.Inheritance;

/**
 * Teacher
 */
public class Teacher extends Person {
    String department;

    Teacher(String name, int age, float salary, String department) {
        super(name, age, salary);
        this.department = department;
    }

    @Override
    void printInfo() {
        super.printInfo();
        System.out.println("Department: " + department + "\n\n");
    }
}